ipfx/ephys_data_set.py

```diff
@@ -81,17 +81,19 @@ def filtered_sweep_table(self,
             mask = st[self.SWEEP_NUMBER] == sweep_number
             st = st[mask]
 
-
         return st
 
-    def get_sweep_number_by_stimulus_names(self, stimulus_names):
+    def get_sweep_number(self, stimuli, clamp_mode=None):
+
+        sweeps = self.filtered_sweep_table(stimuli=stimuli).sort_values(by=self.SWEEP_NUMBER)
 
-        sweeps = self.filtered_sweep_table(
-            stimuli=stimulus_names).sort_values(by=self.SWEEP_NUMBER)
+        if clamp_mode:
+            mask = sweeps[self.CLAMP_MODE] == clamp_mode
+            sweeps = sweeps[mask]
```

Comment:
@sgratiy If you teach `filtered_sweep_table` to allow selection by clamp mode you can pass in both the clamp mode and the simuli and get the result without having to reimplement the logic here.

It would be also nice to check if the given clamp mode exists to detect typos. Or we have, even better, global constants for these strings.

Reply:
This are good suggestions.  Previously, I have stopped short of doing this because I was hesitant of touching `filtered_sweet_table` function used in several places in the code. But, it turned out not difficult in the end. Thnx.